### PR TITLE
refactor(es/ast): change ClassProp definition

### LINKED ISSUE #2516: class with same name static property is wrongly renamed
**Describe the bug**

**Input code**
```js
class A {
  static A = class {}
}
```

and output is
```js
class A1 {
  static A = class {}
}
```

**Config**

```json
{
  "env": {
     "target": "chrome 80"
  }
}
```

**Expected behavior**
Output should be identical to input

**Version**
The version of @swc/core: latest

### LINKED ISSUE #3026: Change definition of ClassProp
### Discussed in https://github.com/swc-project/swc/discussions/3025

<div type='discussions-op-text'>

<sup>Originally posted by **Austaras** December 12, 2021</sup>
Change `key` field to `PropName` and remove `computed`. There's two reasons:
1. ClassProp key and ClassMethod key are both `ClassElementName` in spec, it's better to unify them
图片
图片
2. Current definition may generate invalid state. When `computed` is false, `key` could only be `Number` `String` `BigInt`

Do we need to wait for a new major release to fix this?</div>

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**
make ClassProp key PropName

**BREAKING CHANGE:**
It will change the definition of ast tree

**Related issue (if exists):**
It will close #3026, and as an intended side effect, close #2516 and some unreported decorator issue

### CODE REVIEW

File: crates/swc_ecma_codegen/src/lib.rs [hunk 1]
```diff
@@ -1209,7 +1209,7 @@ where
             space!()
         }
 
-        if n.computed {
+        if n.key.is_computed() {
```
Comment (kdy1) [inline crates/swc_ecma_codegen/src/lib.rs:1212]: You can replace this with `emit!(n.key)`.
The emit method for `PropName` is aware of `Computed`.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Mostly LGTM. Thank you!
Can you rebase this?

**Comment by Austaras** [comment]
Done

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**